runtime/pack/dist/opt/termdebug/plugin/termdebug.vim

```diff
@@ -630,76 +630,84 @@ function! s:OpenHoverPreview(lines, filetype) abort
-            if lw > width
-                let width = lw
-            endif
-            let lines[index] = line
-        endfor
-
-        " Give margin
-        let width += 1
-        let lines = [''] + lines + ['']
-        let height = len(lines)
-
-        " Calculate anchor
-        " Prefer North, but if there is no space, fallback into South
-        let bottom_line = line('w0') + winheight(0) - 1
-        if pos[1] + height <= bottom_line
-            let vert = 'N'
-            let row = 1
-        else
-            let vert = 'S'
-            let row = 0
+      let bufname = nvim_create_buf(v:false, v:true)
+      call nvim_buf_set_lines(bufname, 0, -1, v:true, lines)
+      let pos = getpos('.')
+
+      " Calculate width and height and give margin to lines
+      let width = 0
+      for index in range(len(lines))
+        let line = lines[index]
+        if line !=# ''
+          " Give a left margin
+          let line = ' ' . line
         endif
-
-        " Prefer West, but if there is no space, fallback into East
-        if pos[2] + width <= &columns
-            let hor = 'W'
-            let col = 0
-        else
-            let hor = 'E'
-            let col = 1
+        let lw = strdisplaywidth(line)
+        if lw > width
+          let width = lw
         endif
+        let lines[index] = line
+      endfor
+
+      " Give margin
+      let width += 1
+      let lines = [''] + lines + ['']
+      let height = len(lines)
+
+      " Calculate anchor
+      " Prefer North, but if there is no space, fallback into South
+      let bottom_line = line('w0') + winheight(0) - 1
+      if pos[1] + height <= bottom_line
+        let vert = 'N'
+        let row = 1
+      else
+        let vert = 'S'
+        let row = 0
+      endif
 
-        let float_win_id = nvim_open_win(bufnr, v:true, {
-        \   'relative': 'cursor',
-        \   'anchor': vert . hor,
-        \   'row': row,
-        \   'col': col,
-        \   'width': width,
-        \   'height': height,
-        \ })
+      " Prefer West, but if there is no space, fallback into East
+      if pos[2] + width <= &columns
+        let hor = 'W'
+        let col = 0
+      else
+        let hor = 'E'
+        let col = 1
+      endif
 
-        execute 'noswapfile edit!' bufname
+      let float_win_id = nvim_open_win(bufnr, v:true, {
+            \   'relative': 'cursor',
+            \   'anchor': vert . hor,
+            \   'row': row,
+            \   'col': col,
+            \   'width': width,
+            \   'height': height,
+            \ })
 
-        setlocal winhl=Normal:CursorLine
+      execute 'noswapfile edit!' bufname
+
+      setlocal winhl=Normal:NormalFloat
+      setlocal buftype=nofile nobuflisted bufhidden=wipe nonumber norelativenumber signcolumn=no
+
+      if a:filetype isnot v:null
+        let &filetype = a:filetype
+      endif
+
+      call setline(1, lines)
+      setlocal nomodified nomodifiable
+
+      wincmd p
```

Comment:
set second arg of `win_open_win` to `v:false` instead, to not clobber the user's prevwin. All stuff in between can use api calls with window id.

Reply:
I found that using `nvim_open_win(buf, v:false, ...)` make the floating window disappear (actually nothing happen, no window is shown)

Reply:
I'm using `nvim_win_set_option` to set option like `relativenumber`. However, it does not work for `modified` and `modifiable`...

Reply:
you have to use `nvim_buf_set_option(buf, ...)` for buffer options

Reply:
thanks!

Reply:
> I found that using nvim_open_win(buf, v:false, ...) make the floating window disappear (actually nothing happen, no window is shown)

This shouldn't be the case, there must be a problem with the code below. Possibly some of the autocmds trigger too early.

Reply:
Good catch!
So setting the `{enter}` option of `nvim_open_win` to `v:true` make the cursor automatically enter the floating window.
The effect of moving the cursor automatically trigger `CloseFloatingHoverOnCursorMove` autocommand.

But I think that this option should remain to false, because we don't want the cursor to move in the floating window.

Reply:
d50f7ea
967d6b8
f5d7a9f